keywrap/jwe/keywrapper_jwe.go

```diff
@@ -65,7 +65,11 @@ func (kw *jweKeyWrapper) WrapKeys(ec *config.EncryptConfig, optsData []byte) ([]
 }
 
 func (kw *jweKeyWrapper) UnwrapKey(dc *config.DecryptConfig, jweString []byte) ([]byte, error) {
-	jwe, err := jose.ParseEncrypted(string(jweString))
+	// cf. list of algorithms in func addPubKeys() below
+	key_algorithms := []jose.KeyAlgorithm{jose.RSA_OAEP, jose.RSA_OAEP_256, jose.ECDH_ES_A128KW, jose.ECDH_ES_A192KW, jose.ECDH_ES_A256KW}
+	// accept all algorithms defined in RFC 7518, section 5.1
+	content_enc := []jose.ContentEncryption{jose.A128CBC_HS256, jose.A192CBC_HS384, jose.A256CBC_HS512, jose.A128GCM, jose.A192GCM, jose.A256GCM}
```

Comment:
Are we likely to encounter any of these considering we use public key crypto? I have a few test keys in the imgcrypt project and they are all RSA and EC keys, so don't seem to have this. Otherwise looks good.

https://github.com/containerd/imgcrypt/blob/3ca09a2db5cd722c7548977e9899183ade5245c6/script/tests/test_encryption.sh#L50-L63

Reply:
I honestly don't know. I opted for including every algorithm from go-jose to be safe. Please let me know or amend my change if a shorter list is sufficient.

I figured that a full list isn't making things worse as that mimics the current behavior

Reply:
It looks like there's not even another choice for the content_enc parameter:

```
// The keyAlgorithms and contentEncryption parameters are used to validate the "alg" and "enc"
// header parameters respectively. They must be nonempty, and each "alg" or "enc" header in
// parsed data must contain a value that is present in the corresponding parameter. That
// includes the protected and unprotected headers as well as all recipients. To accept
// multiple algorithms, pass a slice of all the algorithms you want to accept.
```

https://github.com/go-jose/go-jose/blob/dc40081e51fadc58e5c707cc4c9e5aebb7a53571/jwe.go#L131-L135